plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestRegexTable.java

```diff
@@ -0,0 +1,147 @@
+                CREATE TABLE test_regex_data
+                WITH (
+                    format = 'textfile',
+                    textfile_field_separator = 'x',
+                    external_location = '%s')
+                AS SELECT nationkey, name FROM tpch.tiny.nation
+                """.formatted(tableLocation.toUri().toASCIIString());
+        assertUpdate(createTableSql, 25);
+
+        MaterializedResult expected = computeActual("SELECT nationkey, name FROM tpch.tiny.nation");
+        MaterializedResult actual = computeActual("SELECT nationkey, name FROM test_regex_data");
+        assertEqualsIgnoreOrder(actual.getMaterializedRows(), expected.getMaterializedRows());
+
+        // REGEX table over the text file created data
+        createTableSql = """
+                CREATE TABLE test_regex (
+                    nationkey BIGINT,
+                    name VARCHAR)
+                WITH (
+                    format = 'regex',
+                    regex = '(\\d+)x(.+)',
+                    external_location = '%s')
+                """.formatted(tableLocation.toUri().toASCIIString());
+        assertUpdate(createTableSql);
+
+        actual = computeActual("SELECT nationkey, name FROM test_regex");
+        assertEqualsIgnoreOrder(actual.getMaterializedRows(), expected.getMaterializedRows());
+
+        // Verify REGEX read-only is enforced
+        assertQueryFails("INSERT INTO test_regex VALUES (42, 'name')", "REGEX format is read-only");
+
+        // case insensitive
+        assertUpdate("DROP TABLE test_regex");
+        createTableSql = """
+                CREATE TABLE test_regex (
+                    nationkey BIGINT,
+                    name VARCHAR)
+                WITH (
+                    format = 'regex',
+                    regex = '(\\d+)X(.+)',
+                    regex_case_insensitive = true,
+                    external_location = '%s')
+                """.formatted(tableLocation.toUri().toASCIIString());
+        assertUpdate(createTableSql);
+        actual = computeActual("SELECT nationkey, name FROM test_regex");
+        assertEqualsIgnoreOrder(actual.getMaterializedRows(), expected.getMaterializedRows());
+
+        // case-sensitive with no-match
+        assertUpdate("DROP TABLE test_regex");
+        createTableSql = """
+                CREATE TABLE test_regex (
+                    nationkey BIGINT,
+                    name VARCHAR)
+                WITH (
+                    format = 'regex',
+                    regex = '(\\d+)X(.+)',
+                    external_location = '%s')
+                """.formatted(tableLocation.toUri().toASCIIString());
+        assertUpdate(createTableSql);
+        // when the pattern does not match all columns are null
+        actual = computeActual("SELECT nationkey, name FROM test_regex WHERE nationkey is not null and name is not null");
+        assertEquals(actual.getMaterializedRows(), ImmutableList.of());
```

Comment:
I'm working on a PR to get rid of `io.trino.testing.assertions.Assert.assertEquals`. This can be
```java
assertQueryReturnsEmptyResult("SELECT nationkey, name FROM test_regex WHERE nationkey IS NOT NULL AND name IS NOT NULL");
```
Also capitalize SQL keywords